### PR TITLE
OplogToRedis: Run writer routines in parallel for each redis client.

### PR DESCRIPTION
**Pull request by alex-goodisman**
WIP

in main, count through each redis client and make a separate chan and coroutine for clientsCount * ordinalCount different entries. make some changes to aggregation in order to accomodate this.

in the tailer, get the [][]chan and for each incoming publication, first index by the shard ordinal (hash of db name), then iterate through the remaining chans (should be one for each redis client) and publish to all of them. 

at present, the publisher stream still accepts an array, but now it's only getting one (each).

### CODE REVIEW

File: lib/oplog/tail.go [hunk 1]
```diff
@@ -116,7 +116,7 @@ func init() {
 
 // Tail begins tailing the oplog. It doesn't return unless it receives a message
 // on the stop channel, in which case it wraps up its work and then returns.
-func (tailer *Tailer) Tail(out []chan<- *redispub.Publication, stop <-chan bool) {
+func (tailer *Tailer) Tail(out [][]chan<- *redispub.Publication, stop <-chan bool) {
```
Comment (torywheelwright) [inline lib/oplog/tail.go:119]: this 2d array of channels is getting complicated-enough that the semantics aren't obvious. could we document it, or use type aliases, or something similar that would make it easier to understand?

File: main.go [hunk 2]
```diff
@@ -83,42 +83,56 @@ func main() {
...
+
+			redisPubs := make(chan *redispub.Publication, bufferSize)
+			redisPubsAggregationEntry[j] = redisPubs
+
+			stopRedisPub := make(chan bool)
+			stopRedisPubsEntry[j] = stopRedisPub
+
+			waitGroup.Add(1)
+
+			// We crate two goroutines:
```
Comment (torywheelwright) [inline main.go:102]: create

File: main.go [hunk 3]
```diff
@@ -83,42 +83,56 @@ func main() {
 		log.Log.Infow("Initialized connection to Redis", "i", i)
 
 		aggregatedRedisClients[i] = redisClients
+		clientsSize := len(redisClients)
+
+		redisPubsAggregationEntry := make([]chan<- *redispub.Publication, clientsSize)
+		stopRedisPubsEntry := make([]chan bool, clientsSize)
+
+		for j := 0; j < clientsSize; j++ {
```
Comment (torywheelwright) [inline main.go:91]: general comment, this has grown somewhat hard to follow because of the nested arrays and loops, etc. it would be nice to refactor it to be a bit more abstract when we're back on firmer ground.

**Review by torywheelwright — COMMENTED** [review]
general comment, some of this has grown a bit difficult

File: main.go (hunk 2 above)
Comment (alex-goodisman) [reply main.go:102]: lol that was there, i'll do a spelling pass

File: main.go (hunk 3 above)
Comment (alex-goodisman) [reply main.go:91]: I think type aliases will help with this somewhat, but will also add some comments.

**Review by torywheelwright — APPROVED** [review]
Thanks for the understandability changes; they helped a lot

**Comment by aranair** [comment]
(small reminder to bump version before merging)
